src/cpp/session/modules/SessionSource.cpp

```diff
@@ -707,9 +707,19 @@ Error closeDocument(const json::JsonRpcRequest& request,
    // get the path (it's okay if this fails, unsaved docs don't have a path)
    std::string path;
    source_database::getPath(id, &path);
+
+   // retrieve document from source database
+   boost::shared_ptr<source_database::SourceDocument> pDoc(
+               new source_database::SourceDocument());
+   error = source_database::get(id, pDoc);
+   if (error)
+   {
+      LOG_ERROR(error);
+      return error;
```

Comment:
Hah, sorry I made this harder than it needed to be! I think if we hit an error here we shouldn't fire the onDocPendingRemove event (since we have nothing to feed it) but we should still try to remove the doc from the source database (otherwise a corrupt/problematic doc could get stuck in there).